lib/rgeo/shapefile/reader.rb

```diff
@@ -185,6 +185,7 @@ def initialize(path_, opts_ = {}) # :nodoc:
           if ::File.file?(path_ + ".dbf") && ::File.readable?(path_ + ".dbf")
             if ::File.file?(path_ + ".cpg") && ::File.readable?(path_ + ".cpg")
               dbf_encoding_ = ::File.read(path_ + ".cpg")
+              dbf_encoding_ = Encoding.find(dbf_encoding_) rescue nil
```

Comment:
What exception are we expecting here?

Reply:
It will be `ArgumentError` if the encoding name is not one that Ruby recognizes or `TypeError` if we end up passing `nil`.